metadata-ingestion/src/datahub/emitter/rest_emitter.py

```diff
@@ -228,11 +233,15 @@ def emit_mce(self, mce: MetadataChangeEvent) -> None:
         snapshot_fqn = (
             f"com.linkedin.metadata.snapshot.{mce.proposedSnapshot.RECORD_SCHEMA.name}"
         )
+        ensure_mce_has_system_metadata(mce)
         system_metadata_obj = {}
         if mce.systemMetadata is not None:
             system_metadata_obj = {
                 "lastObserved": mce.systemMetadata.lastObserved,
                 "runId": mce.systemMetadata.runId,
+                "lastRunId": mce.systemMetadata.lastRunId,
+                "clientId": mce.systemMetadata.clientId,
+                "clientVersion": mce.systemMetadata.clientVersion,
```

Comment:
if we emit these fields to a GMS instance that doesn't have the model changes, does it silently drop them or does it throw an error? if it's the latter, we need this setting to be configurable

Reply:
moved to using properties instead as suggested by Shirshanka.